monitor.go

```diff
@@ -588,7 +588,6 @@ func runMonitor() {
 			promPushMetric.Observe(time.Since(pushStartTime).Seconds())
 
 			log.Infof("Test successful")
-			duration = 2 * time.Minute
```

Comment:
This isn't a sufficient change, as we change the duration in other places. You'll need to set it to this: `duration, err := time.ParseDuration(*testInterval)`

Reply:
@josephschorr You are already defining that line here in the main loop so no need to define it again: https://github.com/coreos/registry-monitor/blob/master/monitor.go#L495

You change duration if it fails to pull image : https://github.com/coreos/registry-monitor/blob/master/monitor.go#L537 and if it fails to push image : https://github.com/coreos/registry-monitor/blob/master/monitor.go#L575 , I think to test again pull and push. But why change duration : https://github.com/coreos/registry-monitor/blob/master/monitor.go#L591 defined by user : https://github.com/coreos/registry-monitor/blob/master/monitor.go#L495 if test is ok ?

I tested in my production and removing that line : https://github.com/coreos/registry-monitor/blob/master/monitor.go#L591 as proposed in my pull request make you code works I think as intended.


Thanks in advance for your input and if you could provide me input on my other pull request : https://github.com/coreos/registry-monitor/pull/22, I would be really thankful ! :D

Reply:
Right, but then we need to change it *back* to the user-specified duration once the operation has succeeded, hence the need to add something back here.